src/screens/ImportWalletScreen.tsx

```diff
@@ -100,7 +113,109 @@ export const ImportWalletScreen = (props: Props) => {
     }
   };
 
+  const saveARWallet = async (pubKeyHex: string, mnemonic: string = text) => {
+    try {
+      const wallet = new HDSegwitP2SHArWallet();
+      wallet.setMnemonic(mnemonic);
+      wallet.addPublicKey(pubKeyHex);
+      await wallet.generateAddresses();
+      await wallet.fetchBalance();
+      await wallet.fetchTransactions();
+      if (wallet.getBalance() > 0 || wallet.getTransactions().length !== 0) {
+        wallet.setLabel(label);
+        saveWallet(wallet);
+        showSuccessImportMessageScreen();
+      } else {
+        showErrorMessageScreen();
+      }
+    } catch (error) {
+      showErrorMessageScreen();
+    }
+  };
+
+  const saveAIRWallet = async (publicKey: string, recoveryKey: string, mnemonic: string = text) => {
+    try {
+      const wallet = new HDSegwitP2SHAirWallet();
+      wallet.setMnemonic(mnemonic);
```

Comment:
setMnemonic and addPublicKey are prepared to throw errors if invalid, you should use them at every step of importing wallets so the users gets instant feedback if mnemonic or public key is wrong. Check how this works at wallet creation